lib/webpush/request.rb

```diff
@@ -36,17 +44,26 @@ def headers
       headers["Content-Type"] = "application/octet-stream"
       headers["Ttl"]          = ttl
 
-      if encrypted_payload?
+      if @payload.has_key?(:server_public_key)
         headers["Content-Encoding"] = "aesgcm"
-        headers["Encryption"] = "salt=#{salt_param}"
-        headers["Crypto-Key"] = "dh=#{dh_param}"
+        headers["Encryption"] = "keyid=p256dh;salt=#{salt_param}"
+        headers["Crypto-Key"] = "keyid=p256dh;dh=#{dh_param}"
       end
 
-      headers["Authorization"] = "key=#{api_key}" if api_key?
+      vapid_headers = build_vapid_headers
+      headers["Authorization"] = vapid_headers["Authorization"]
```

Comment:
Good point. It was perhaps aggressive of me to remove this functionality so I've added an item to my checklist to reinstate.